runtime/gc_glue_java/CompactSchemeFixupObject.cpp

```diff
@@ -140,6 +147,7 @@ MM_CompactSchemeFixupObject::fixupObject(MM_EnvironmentStandard *env, omrobjectp
 		fixupMixedObject(objectPtr);
 		break;
 	case GC_ObjectModel::SCAN_CONTINUATION_OBJECT:
+		addContinuationObjectInList(env, objectPtr);
```

Comment:
This does not seem right (in any of moving GCs that you've added to).
 OwnableSyncronizer is not a clearable root (except for Concurrent Global Marking) . We rebuild the list from scratch here and we need to add it the new list whenever we visit it via fixup walk. 
 Continuation is a clearable root. We don't need to rebuild the list, but link pointer just needs to be fixed up. It's done in a separate pass, like MM_CompactSchemeFixupRoots::fixupUnfinalizedObjects, since Unfinalized is clearable, too. (perhaps, we could may do it in the next step in fixupContinuationObject, but let's just do the same as Unfinalized)

Reply:
regularly we need to do it in clearable phase, because we need to clean-up "dead" continuation Object, but for compaction scheme, there is no dead Object, so we can piggyback fixup cycle to rebuild it similar with OwnableSyncronizer.

Reply:
Agree, that would be more efficient, although I was looking more for code uniformity for clearable roots. Probably even more efficient would but just to fix the existing link pointer in fixupContinuationObject. Ok, lets leave it then as-is to minimize last moment changes.